### ISSUE TITLE
livox lidar

### ISSUE THREAD

**Issue by lijian636** [wh200720041/ssl_slam#22]
Can this algorithm be used on livox lidar?

**Comment by Lollipopkkk** [comment]
I have a  lidar producted by neuvition,so i try to use this lidar to make slam by this algoithm.If you do it,please share you experience,thx

**Comment by wdczz** [comment]
I also use neuvition to slam.If  you do it, please share me, too TUT

**Comment by Lollipopkkk** [comment]
> I also use neuvition to slam.If you do it, please share me, too TUT我也用neuviting来猛烈抨击。如果你这样做，请分享我，太嘟
I think it can't be used by this algoithm.
I'm a chinese student,you can use the email address xkg1234@qq.com to send message to me.

**Comment by Lollipopkkk** [comment]
> I also use neuvition to slam.If you do it, please share me, too TUT
I can't think it can be used  by this algoithm,and my email address is xkg1234@qq.com.

**Comment by mbiselx** [comment]
I've been testing it with the Livox HAP. 
It required a few changes to get it to work (you'll find them on [my fork](https://github.com/mbiselx/ssl_slam), but it's mainly a question of changing all point types from `PointXYZRGB` to `PointXYZI`). 
The results are still not very good so far. I'm guessing this is due to the non-repetitve scanning pattern used in this lidar (and most other Livox lidars), as I'm also testing it with a velarray M1600 which does have a repetitive scanning pattern and which is gives decent results.

**Comment by Lollipopkkk** [comment]
> I've been testing it with the Livox HAP. It required a few changes to get it to work (you'll find them on [my fork](https://github.com/mbiselx/ssl_slam), but it's mainly a question of changing all point types from `PointXYZRGB` to `PointXYZI`). The results are still not very good so far. I'm guessing this is due to the non-repetitve scanning pattern used in this lidar (and most other Livox lidars), as I'm also testing it with a velarray M1600 which does have a repetitive scanning pattern and which is gives decent results.
you're right,Livox(DJI) and neuvition that i used is the non-repetitive scanning pattern lidar,they aren't as same as the repetitive lidar. you can use RS-lidar-M1 or L515 to implement this algorithm.
in fact,this program looks a lot like floam.do you know livox-loam?

**Comment by wdczz** [comment]
Actually, I give up this algorithm with my neuvition , I try running fastlio2 with livoxmid40.Sometimes,there are problems bacause of lidar itself. TUT
